### DISCUSSION TITLE
Pulumi Plugin Installation

### DISCUSSION THREAD

**Discussion by anitalySuper** [pulumi/pulumi#16772] [General]
Hello, I'm trying to understand how a plugin version is determined. I'm looking to use pulumi-snowflake v0.54.0, which I have installed using poetry, and this is typically specified in my pyproject.toml/poetry.lock file. However, when I run pulumi stack select (stack_name), it automatically installs v0.56.0, which is the latest version.

Is there a way to get the plugin version to reference what is installed as a package?

**Comment by Frassle** [comment]
What does `pulumi about` report for your project?
It might be it's trying to install 0.56.0 but will also install and use 0.54.0 when running the actual program.

**Reply by anitalySuper** [reply]
This is the output:

CLI          
Version      3.111.1
Go Version   go1.22.1
Go Compiler  gc

Host     
OS       debian
Version  11.9
Arch     x86_64

Backend        
Name           pulumi.com
URL            https://app.pulumi.com
User           Unknown
Organizations  
Token type     personal

**Reply by anitalySuper** [reply]
It seems to be installing 0.56.0 and using that version because 0.56.0 has a bunch of breaking changes that are showing up.

**Reply by Frassle** [reply]
> I'm trying to understand how a plugin version is determined

So the version is initially determined by looking at your project dependencies. `about` should show that if you run it from the same directory as your python project. 

But the version is also part of the SDK and sent to the engine to confirm the version to use. If pulumi-snowflake v0.54.0 is installed but it's using v0.56.0 it's possibly a bug in that SDK that the version hasn't been embedded in the SDK and so no version is present and the engine is falling back to using the latest version (which is 0.56.0)

**Reply by anitalySuper** [reply]
I'm running all of this in a docker container, where v0.54.0 is installed. How can I go about confirming whether this is a bug or not?

**Reply by anitalySuper** [reply]
Hi @Frassle, one thing I didn't specify earlier is that I'm using Pulumi yaml. Whenever Pulumi runs with a yaml set up, do I always need specify pulumi install plugin?

**Reply by Frassle** [reply]
Ah! You probably want to look at the defaultProvider from https://www.pulumi.com/docs/languages-sdks/yaml/yaml-language-reference/. 

YAML doesn't have SDKs so you need to specify the version yourself.
